File: backend/lib/news-domain/src/jvmMain/kotlin/me/ricoschulz/newsDomain/entity/Article.kt
package me.ricoschulz.newsDomain.entity

import java.util.UUID
import me.ricoschulz.newsDomain.valueObject.Product
import me.ricoschulz.newsDomain.valueObject.Provider

data class Article(
    val id: String = UUID.randomUUID().toString(),
    val provider: Provider,
    val title: String,
    val subTitle: String? = null,
    val img: String,
    val link: String,
    val description: String? = null,
    val date: String,
    val products: List<Product> = listOf(),
    val tags: List<String> = listOf()
) {
    val hash = UUID.nameUUIDFromBytes(title.toByteArray()).toString()
}

File: app/src/article/entities/Article.ts
import Provider from "article/value-objects/Provider";
import Product from "./Product";

type Article = {
    id: string
    title: string
    subTitle?: string
    date: string
    link: string
    image: string

    description?: string
    provider: Provider
    products: Product[]
}

export default Article;
